pages/agent/v3/hooks.md

```diff
@@ -136,17 +136,19 @@ they are run as part of each job:
 | `pre-command`   | <span class="add-icon-agent">Agent</span><br /><span class="add-icon-repository">Repository</span><br /><span class="add-icon-plugin">Plugin (non-vendored)</span><br /><span class="add-icon-plugin">Plugin (vendored)</span> | Runs before the build command |
-| `command`       | <span class="add-icon-plugin">Plugin (non-vendored)</span><br /><span class="add-icon-plugin">Plugin (vendored)</span><br /><span class="add-icon-repository">Repository</span><br /><span class="add-icon-agent">Agent</span> | Overrides the default command running behavior. If multiple command hooks are found, only the first will be run. |
+| `command`       | <span class="add-icon-plugin">Plugin (non-vendored)</span><br /><span class="add-icon-plugin">Plugin (vendored)</span><br /><span class="add-icon-repository">Repository</span><br /><span class="add-icon-agent">Agent</span> | Overrides the default command running behavior. If multiple command hooks are found, only the first will be run.[^plugin-exception] |
 | `post-command`  | <span class="add-icon-agent">Agent</span><br /><span class="add-icon-repository">Repository</span><br /><span class="add-icon-plugin">Plugin (non-vendored)</span><br /><span class="add-icon-plugin">Plugin (vendored)</span> | Runs after the command. |
 | `pre-artifact`  | <span class="add-icon-agent">Agent</span><br /><span class="add-icon-repository">Repository</span><br /><span class="add-icon-plugin">Plugin (non-vendored)</span><br /><span class="add-icon-plugin">Plugin (vendored)</span> | Runs before artifacts are uploaded, if an artifact upload pattern was defined for the job. |
 | `post-artifact` | <span class="add-icon-agent">Agent</span><br /><span class="add-icon-repository">Repository</span><br /><span class="add-icon-plugin">Plugin (non-vendored)</span><br /><span class="add-icon-plugin">Plugin (vendored)</span> | Runs after artifacts have been uploaded, if an artifact upload pattern was defined for the job. |
 | `pre-exit`      | <span class="add-icon-agent">Agent</span><br /><span class="add-icon-repository">Repository</span><br /><span class="add-icon-plugin">Plugin (non-vendored)</span><br /><span class="add-icon-plugin">Plugin (vendored)</span> | Runs before the job finishes. Useful for performing cleanup tasks. |
 {: class="table table--no-wrap"}
 
+[^plugin-exception]: For legacy compatibility reasons, this restriction does not apply to plugins. All checkout or command hooks provided by plugins will be run in the order of plugins, which means that multiple checkout hooks and command hooks can run. However, any checkout hooks and command hooks provided by plugins will prevent any repository or agent hooks of the same type from running.
```

Comment:
It seems this does not render as a footnote. Is there a better way to format this?

Reply:
Interesting! I haven't seen footnotes used very heavily in dev docs, but I wonder if it's worth exploring. From what I can see, our tooling does support this syntax but it's not enabled. For now, I suggest adding a new section after "Creating job lifecycle hooks:"

```md
### Hook exception for plugins

As of Agent v3.15.0, only the first checkout and command hooks will run if there are multiple found. However, there is an exception with plugins for legacy compatibility. All checkout or command hooks provided by plugins will run in the order the plugins are specified, meaning multiple checkout and command hooks can run. Note that checkout hooks and command hooks provided by plugins will prevent any repository or agent hooks of the same type from running.
```

And linking to it in the table like:

```md
..., only the first will be run (see [Hook exception for plugins](#job-lifecycle-hooks-hook-exception-for-plugins))
```

What do you think?

Reply:
Sounds good to me.